pkgs/development/interpreters/python/mk-python-derivation.nix

```diff
@@ -179,7 +176,25 @@ let
     # If given use the specified checkPhase, otherwise use the setup hook.
     # Longer-term we should get rid of `checkPhase` and use `installCheckPhase`.
     installCheckPhase = attrs.checkPhase;
-  }));
+  } // lib.optionalAttrs (python.stdenv.hostPlatform != python.stdenv.buildPlatform)
+    (with python.stdenv.hostPlatform; let
+    # See: https://github.com/python/cpython/blob/e488e300f5c01289c10906c2e53a8e43d6de32d8/configure.ac#L724
+    pythonArch =
+      if isAarch32 then
+        if parsed.cpu.significantByte.name == "littleEndian" then "arm" else "armeb"
+      else parsed.cpu.name;
+    pythonHostPlatform = "${pythonArch}-${parsed.kernel.name}-${parsed.abi.name}";
+  in {
+    # Python packages built through cross-compilation are always for the host platform.
+    disallowedReferences = [ python.pythonForBuild ];
+    # Make distutils compile extension modules against host Python
+    _PYTHON_HOST_PLATFORM = pythonHostPlatform;
```

Comment:
It was suggested in https://github.com/NixOS/nixpkgs/issues/53320#issuecomment-572982797 to set this in the setup hook. I'm not sure which is more appropriate.

Reply:
This is best put in `pkgs/development/interpreters/python/setup-hook.sh`. That contains interpreter-specific configuration when using the interpreter in a build.

Reply:
I'm working on adding this to the setup hook, but these variables need to only be set for host Python and I can't think of a way to detect that in the setup hook. Setting them if Python is cross-compiled would probably work in practice, but that seems kind of hacky.